### DISCUSSION TITLE
PJ-1203A power meter low level commands

### DISCUSSION THREAD

**Discussion by maxrudiy** [arendst/Tasmota#21971] [Q&A] [answered]
Hello, I have this Tuya? power meter, i want to change update frequency, console looks like this

> 07:36:16.342 ZigbeeZCLSend device: 0xBFFB, endpoint:1, cluster:0xEF00, cmd:0x0B, send:"0200"
> 07:36:16.343 ZIG: ZbEZSPSend 340000FBBF040100EF010100000000BE010510BE0B0200
> 07:36:16.345 ZIG: ZbZCLRawReceived: {"0xBFFB":{"EF00/0279":85,"Endpoint":1,"LinkQuality":94}}
> 07:36:16.350 MQT: tele/tasmota_2D4030/SENSOR = {"ZbReceived":{"0xBFFB":{"Device":"0xBFFB","Name":"PJ-1203A","EF00/0279":85,"Endpoint":1,"LinkQuality":94}}}
> 07:36:16.387 ZIG: {"ZbEZSPReceived":"3400006C"}
> 07:36:16.438 ZIG: {"ZbEZSPReceived":"3F0000FBBF040100EF0101000000006C010000"}
> 07:36:16.488 ZIG: {"ZbEZSPReceived":"450000040100EF0101000100009DC4CDFBBFFFFF0D09BF0200156F0200040000133B"}
> 07:36:16.490 ZIG: {"ZbZCLReceived":{"groupid":0,"clusterid":"0xEF00","srcaddr":"0xBFFB","srcendpoint":1,"dstendpoint":1,"wasbroadcast":0,"LinkQuality":94,"securityuse":0,"seqnumber":157,"fc":"0x09","frametype":1,"direction":1,"disableresp":0,"manuf":"0x0000","transact":191,"cmdid":"0x02","payload":"00156F0200040000133B"}}
> 07:36:16.492 ZigbeeZCLSend device: 0xBFFB, endpoint:1, cluster:0xEF00, cmd:0x0B, send:"0200"
> 07:36:16.493 ZIG: ZbEZSPSend 340000FBBF040100EF010100000000BF010510BF0B0200
> 07:36:16.495 ZIG: ZbZCLRawReceived: {"0xBFFB":{"EF00/026F":4923,"Endpoint":1,"LinkQuality":94}}
> 07:36:16.500 MQT: tele/tasmota_2D4030/SENSOR = {"ZbReceived":{"0xBFFB":{"Device":"0xBFFB","Name":"PJ-1203A","EF00/026F":4923,"Endpoint":1,"LinkQuality":94}}}

Tried something like this etc
TuyaSend2 129, 5
ZbSend {"device":"PJ-1203A","endpoint":"1","send":"EF00!0B/0200"}
ZbSend {"device":"PJ-1203A","endpoint":"1","send":"EF00!0481/000000005"}
ZbSend {"device":"PJ-1203A","endpoint":"1","send":"EF00!0281/000000005"}

But I don't have any response from device after this commands and i don't understand how to send dpid with ZbSend command. 
Maybe I should connect PJ-1203A to native Tuya SmartLife, change settings there and then reconnect to Tasmota Zigbee Hub/ Because I don't have other ideas...

![258651940-df4090a1-1afb-4007-a302-9664a79fd811](https://github.com/user-attachments/assets/118c1c10-0df1-4685-97ba-b0fd61068085)

**Comment by s-hadinger** [comment]
Oh Tuya and Zigbee…

Cluster is `EF00` and attribute is decomposed like this `xxyy` where `xx` is data type and `yy` the dpid in hex. 

Types are:
```
 // Mapping of Tuya type with internal mapping
  // 0x00 - Zoctstr (len N)
  // 0x01 - Ztuya1 (len 1) - equivalent to Zuint8 without invalid value handling
  // 0x02 - Ztuya4 (len 4) - equivalent to Zint32 in big endian and without invalid value handling
  // 0x03 - Zstr (len N)
  // 0x04 - Ztuya1 (len 1)
  // 0x05 - Ztuya4u (len 1/2/4)
```

Then just use simple "write" command.

**Reply by s-hadinger** [reply, marked as answer]
Example 

`ZbSend {"device":"0xBFFB","Write":{"EF00/0281":60}}`

**Reply by maxrudiy** [reply]
Wow! wow! it works perfectly! Thank you! I searched internet whole day yesterday and came nowhere, You are my savior:)

**Reply by s-hadinger** [reply]
If you are courageous enough, you can write a plug-in file for this device to put actual names on all dpid attributes. 
See https://tasmota.github.io/docs/Zigbee/#definition-file
